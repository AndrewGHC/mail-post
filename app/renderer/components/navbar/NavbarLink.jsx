import React from 'react';
import PropTypes from 'prop-types';
import { MenuItem } from 'material-ui/Menu';
import { ListItemIcon, ListItemSecondaryAction } from 'material-ui/List';
import { Link } from '../lib';

const NavbarLink = ({
  to,
  text,
  renderIcon,
  ...rest
}) => (
  <Link to={to} {...rest} >
    <MenuItem>
      {text}
      <ListItemSecondaryAction>
        <ListItemIcon>
          {renderIcon()}
        </ListItemIcon>
      </ListItemSecondaryAction>
    </MenuItem>
  </Link>
);

NavbarLink.propTypes = {
  to: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  renderIcon: PropTypes.func.isRequired,
};

export default NavbarLink;
