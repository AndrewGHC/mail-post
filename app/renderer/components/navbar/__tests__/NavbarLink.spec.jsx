import React from 'react';
import { shallow } from 'enzyme';
import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';

import NavbarLink from '../NavbarLink';

describe('NavbarLink', () => {
  let props;
  beforeEach(() => {
    props = {
      to: 'a route',
      text: 'name',
    };
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<NavbarLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('when renderIcon is passed, renders prop', () => {
    const TestComponent = () => <div />;
    const wrapper = shallow(
      <NavbarLink
        {...props}
        renderIcon={() => <TestComponent />}
      />,
    );
    expect(wrapper.dive().find(TestComponent).exists()).toBe(true);
  });

  it('when renderIcon is NOT passed, renders KeyboardArrowRightIcon', () => {
    const wrapper = shallow(<NavbarLink {...props} />);
    expect(wrapper.dive().find(KeyboardArrowRightIcon).exists()).toBe(true);
  });
});
