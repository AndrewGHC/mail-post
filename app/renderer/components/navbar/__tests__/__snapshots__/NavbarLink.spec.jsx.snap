// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink matches snapshot 1`] = `
<Link
  to="a route"
>
  <WithStyles(MenuItem)>
    name
    <WithStyles(ListItemSecondaryAction)>
      <WithStyles(ListItemIcon) />
    </WithStyles(ListItemSecondaryAction)>
  </WithStyles(MenuItem)>
</Link>
`;
